### PR TITLE
coap-gateway.md: update deployment doc based on yaml config.

### PR DESCRIPTION
**Pull request by Trishia**
@ondrejtomcik @jkralik
I newly updated this guide based on yaml config as well as making certificates, getting/editing config file, running image without source download & build. Review and then give me comments. If good, approve plz.

### CODE REVIEW

File: docs/guide/deployment/coap-gateway.md [hunk 1]
```diff
@@ -24,17 +22,9 @@ TCP connection to the OCF Native Cloud is by its nature stateful. The OCF CoAP G
...
-
-## API
+### API
 
-Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org/specs/OCF_Device_To_Cloud_Services_Specification_v2.2.0.pdf)
-
-### Commands
+CoAP APIs of the Cloud Service are defined in [OCF Device To Cloud Services Specification](https://openconnectivity.org/specs/OCF_Device_To_Cloud_Services_Specification_v2.2.3.pdf).
 
 - POST /oic/sec/account - sign up the device with authorization code
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: please move this to component documentation -> coap gateway.

File: docs/guide/deployment/coap-gateway.md [hunk 2]
```diff
@@ -1,6 +1,4 @@
-# COAP gateway
-
-## Description
+# CoAP Gateway
 
 OCF Servers / Clients communicate over TCP / UDP using the CoAP application protocol. Communication within the OCF Native Cloud shouldn't be restricted to the CoAP protocol, implementation should allow the use of whatever protocol might be introduced in the future. That's why the gateway is the access point for CoAP over TCP, and further communication is OCF Native Cloud specific.
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: CoAP Gateway provides API for devices and clients following the [Open Connectivity Foundation](ocf web) specification.

File: docs/guide/deployment/coap-gateway.md [hunk 3]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+	plgd/coap-gateway:latest --config=/data/coap-gateway.yaml
+```
+
+## YAML Configuration
+### Logging
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `log.debug` | bool | `Set to true if you would like to see extra information on logs.` | `false` |
+| `log.dumpCoapMessages` | bool | `Set to true if you would like to dump coap messages on logs.` | `false` |
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: ... if you would like to dump raw messages.

File: docs/guide/deployment/coap-gateway.md [hunk 4]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+
+### CoAP API
+CoAP API of Coap Gateway Service as defined in [uri](https://github.com/plgd-dev/cloud/blob/docs-deployment-grpc-gateway/coap-gateway/uri/uri.go) and [swagger](https://raw.githubusercontent.com/openconnectivityfoundation/cloud-services/master/swagger2.0/oic.wk.rd.swagger.json).
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: CoAP API as specified in the [Open Connectivity Foundation - Device to Cloud Services](https://openconnectivity.org/specs/OCF_Device_To_Cloud_Services_Specification_v2.2.3.pdf).

File: docs/guide/deployment/coap-gateway.md [hunk 5]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+| ---------- | -------- | -------------- | ------- |
+| `log.debug` | bool | `Set to true if you would like to see extra information on logs.` | `false` |
+| `log.dumpCoapMessages` | bool | `Set to true if you would like to dump coap messages on logs.` | `false` |
+
+### CoAP API
+CoAP API of Coap Gateway Service as defined in [uri](https://github.com/plgd-dev/cloud/blob/docs-deployment-grpc-gateway/coap-gateway/uri/uri.go) and [swagger](https://raw.githubusercontent.com/openconnectivityfoundation/cloud-services/master/swagger2.0/oic.wk.rd.swagger.json). 
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `api.coap.address` | string | `Listen specification <host>:<port> for coap client connection.` | `"0.0.0.0:9100"` |
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: bad example - 9100 -> 5684

File: docs/guide/deployment/coap-gateway.md [hunk 6]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+| `api.coap.externalAddress` | string | `External address including public domain/IP for coap client connection.` | `"0.0.0.0:9100"` |
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: use some random ip address, not 0.0.0.0

File: docs/guide/deployment/coap-gateway.md [hunk 7]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+| `api.coap.maxMessageSize` | int | `Max message size which can be send/receive via coap.` | `262144` |
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: unit?

File: docs/guide/deployment/coap-gateway.md [hunk 8]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+| ---------- | -------- | -------------- | ------- |
+| `clients.eventBus.goPoolSize` | int | `Number of routines to process events in projection.` | `16` |
+| `clients.eventBus.nats.url` | string | `URL to nats messaging system.` | `"nats://localhost:4222"` |
+| `clients.eventBus.nats.tls.caPool` | string | `root certificate the root certificate in PEM format.` |  `""` |
+| `clients.eventBus.nats.tls.keyFile` | string | `File name of private key in PEM format.` | `""` |
+| `clients.eventBus.nats.tls.certFile` | string | `File name of certificate in PEM format.` | `""` |
+| `clients.eventBus.nats.tls.useSystemCAPool` | bool | `If true, use system certification pool.` | `false` |
+
+### Authorization Server Client
+Authorization server client configuration to connect internally.
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: Somewhere you escape it with backshlash, somewhere not. Please unify.

File: docs/guide/deployment/coap-gateway.md [hunk 9]
```diff
@@ -48,56 +38,233 @@ Follow [OCF Device To Cloud Services Specification](https://openconnectivity.org
...
+| `clients.authorizationServer.grpc.tls.certFile` | string | `File path to certificate in PEM format.` | `""` |
+| `clients.authorizationServer.grpc.tls.useSystemCAPool` | bool | `If true, use system certification pool.` | `false` |
+| `clients.authorizationServer.grpc.keepAlive.time` | string | `After a duration of this time if the client doesn't see any activity it pings the server to see if the transport is still alive.` | `10s` |
+| `clients.authorizationServer.grpc.keepAlive.timeout` | string | `After having pinged for keepalive check, the client waits for a duration of Timeout and if no activity is seen even after that the connection is closed.` | `20s` |
+| `clients.authorizationServer.grpc.keepAlive.permitWithoutStream` | bool | `If true, client sends keepalive pings even with no active RPCs. If false, when there are no active RPCs, Time and Timeout will be ignored and no keepalive pings will be sent.` | `false` |
+
+### OAuth2.0 Service Client
+>Configured OAuth2.0 client is used by internal service to request a token used to authorize all calls they execute against the plgd API Gateways.
```
Comment (ondrejtomcik) [inline docs/guide/deployment/coap-gateway.md]: against other plgd APIs.

File: docs/guide/deployment/coap-gateway.md (hunk 8 above)
Comment (Trishia) [reply docs/guide/deployment/coap-gateway.md]: @ondrejtomcik 
What do you mean `backslash` ? Give me more detail.

Comment (ondrejtomcik) [reply docs/guide/deployment/coap-gateway.md]: Formatting with ‘ character.

File: docs/guide/deployment/coap-gateway.md (hunk 2 above)
Comment (Trishia) [reply docs/guide/deployment/coap-gateway.md]: @ondrejtomcik 
Just replace old description with your comments ? or leave old description starting with `OCF Servers / Clients ...` ?

File: docs/guide/deployment/coap-gateway.md (hunk 1 above)
Comment (Trishia) [reply docs/guide/deployment/coap-gateway.md]: Do I have to move only CoAP APIs to component documents ? How about `Validation` section ? leave it ?

File: docs/guide/deployment/coap-gateway.md (hunk 8 above)
Comment (Trishia) [reply docs/guide/deployment/coap-gateway.md]: I don't use backslash character with single quotation mark together on any documents. uhmm...  I'm using goland for editing md files on Mac. Is it has some problem in format compatibility ? Cloud you give some example that captures screen shot ?

**Comment by Trishia** [comment]
@ondrejtomcik 
Do you think we document about oauth-server for mock test in docs/guide/deployment ?

**Comment by ondrejtomcik** [comment]
> @ondrejtomcik 
> 
> Do you think we document about oauth-server for mock test in docs/guide/deployment ?

It's not used only in test but also in bundle. Yes I would document it as well. But with note that it's just for tests, not suited for production.

File: docs/guide/deployment/coap-gateway.md (hunk 2 above)
Comment (ondrejtomcik) [reply docs/guide/deployment/coap-gateway.md]: Change to what I wrote only. Doc which is there currently is pretty old one.

File: docs/guide/architecture/component-overview.md [hunk 10]
```diff
@@ -9,6 +9,22 @@ The CoAP gateway acts act as a CoAP Client, communicating with IoT devices - CoA
 ![L3](/img/diagrams/component-coapgateway.svg =600x)
 :::
 
+### APIs
+
+CoAP APIs of the Cloud Service are defined in [OCF Device To Cloud Services Specification](https://openconnectivity.org/specs/OCF_Device_To_Cloud_Services_Specification_v2.2.3.pdf).
+
+- POST /oic/sec/account - sign up the device with authorization code
```
Comment (ondrejtomcik) [inline docs/guide/architecture/component-overview.md:38]: Move this to architecture, component overview, coapgw.

**Comment by Trishia** [comment]
> > @ondrejtomcik
> > Do you think we document about oauth-server for mock test in docs/guide/deployment ?
> 
> It's not used only in test but also in bundle. Yes I would document it as well. But with note that it's just for tests, not suited for production.

@ondrejtomcik 
Do I create doc for oauth-server on the test/oauth-server as a README.md ? Not in docs/guide/deployment.

File: docs/guide/architecture/component-overview.md (hunk 10 above)
Comment (Trishia) [reply docs/guide/architecture/component-overview.md:38]: Yes, I already moved that APIs to architecture/component-overview#CoAP Gateway as you see.

**Review by ondrejtomcik — APPROVED** [review]

File: docs/guide/deployment/coap-gateway.md (hunk 2 above)
Comment (Trishia) [reply docs/guide/deployment/coap-gateway.md]: @ondrejtomcik 
Just moment before approving  plz. Doc is currently Ok ?

File: docs/guide/architecture/component-overview.md (hunk 10 above)
Comment (ondrejtomcik) [reply docs/guide/architecture/component-overview.md:38]: But you didn't remove it from here :)

File: docs/guide/deployment/coap-gateway.md (hunk 1 above)
Comment (ondrejtomcik) [reply docs/guide/deployment/coap-gateway.md]: Validation move as well.

File: docs/guide/deployment/coap-gateway.md (hunk 2 above)
Comment (ondrejtomcik) [reply docs/guide/deployment/coap-gateway.md]: OK

File: docs/guide/deployment/coap-gateway.md (hunk 8 above)
Comment (ondrejtomcik) [reply docs/guide/deployment/coap-gateway.md]: It's OK now. Not sure where I saw the difference.

File: docs/guide/architecture/component-overview.md (hunk 10 above)
Comment (Trishia) [reply docs/guide/architecture/component-overview.md:38]: This part is originally from docs/guide/deployment/coap-gateway.md. So I moved this part to docs/guide/archtecture/component-overview.md#coap-gateway.

**Pull request merged**